server/lib/modules/ose-installer/launch.rb

```diff
@@ -0,0 +1,181 @@
+    opts.on('-x', '--subdomain_name [STRING]', String, 'subdomain name') { |x| options[:subdomain_name] = x }
+    opts.on('-u', '--username [STRING]', String, 'ssh username') { |u| options[:username] = u }
+    opts.on('-k', '--ssh_key [STRING]', String, 'ssh key file') { |k| options[:ssh_key] = k }
+    opts.on('-o', '--ose_user [STRING]', String, 'OSE login username') { |o| options[:ose_user] = o }
+    opts.on('-w', '--ose_password [STRING]', String, 'OSE login password') { |w| options[:ose_password] = w }
+    opts.on('-l', '--log [STRING]', String, 'log file path') { |l| options[:log] = l }
+    opts.on('-v', '--verbose', 'Run verbosely') { |v| options[:verbose] = v }
+  end.parse!
+
+  options
+end
+
+def write_inventory(opts, inventory_file_name="#{OUTPUT_DIR}/ansible.hosts", template_file_name="templates/ansible.hosts.template")
+  template = File.read(template_file_name)
+  template = template.gsub(/<ssh_user>/, opts[:username])
+  if not opts[:username].eql? "root"
+    template = template.gsub(/#ansible_sudo=true/, "ansible_sudo=true")
+  end
+  
+  masters_list = opts[:masters].join("\n")
+  template = template.gsub(/<master_nodes>/, masters_list)
+
+  nodes_list = opts[:nodes].join("\n")
+  template = template.gsub(/<worker_nodes>/, nodes_list)
+
+  # docker storage specifics
+  template = template.gsub(/<docker_storage>/, opts[:docker_storage])
+  template = template.gsub(/<docker_volume>/, opts[:docker_volume])
+  template = template.gsub(/<docker_registry_host>/, opts[:docker_registry_host])
+  template = template.gsub(/<docker_registry_path>/, opts[:docker_registry_path])
+
+  # ose post installation specifics
+  template = template.gsub(/<ose_user>/, opts[:ose_user])
+  template = template.gsub(/<ose_password>/, opts[:ose_password])
+
+  unless Dir.exists?(OUTPUT_DIR)
+    FileUtils.mkdir_p(OUTPUT_DIR)
+  end
+
+  File.open(inventory_file_name, 'w') {|file| file.puts template }
+
+  inventory_file_name
+end
+
+def write_atomic_installer_answer_file(opts, answer_file_name="#{OUTPUT_DIR}/atomic-openshift-installer.answers.cfg.yml", template_file_name="templates/atomic-openshift-installer.answers.cfg.yml.template")
+  template = File.read(template_file_name)
+
+  node_entries = nil
+  opts[:nodes].each do |n|
+    stdout = `getent hosts #{n}`
+    if stdout.empty?
+      p "Cannot resolve ip for #{n}, skipping"
+      next
+    else
+      ip = stdout.split("\n").first.split("  ").first
+    end
+
+    entry = "- connect_to: #{n}\n  hostname: #{n}\n  ip: #{ip}\n  node: true\n  public_hostname: #{n}\n  public_ip: #{ip}"
+    if node_entries.nil?
+      node_entries = entry
+    else
+      node_entries = node_entries + entry
+    end
+  end
+
+  template = template.gsub(/<node_entries>/, node_entries) if not node_entries.nil? 
+
+  master_entries = nil
+  opts[:masters].each do |m|
+    stdout = `getent hosts #{m}`
+    if stdout.empty?
+      p "Cannot resolve ip for #{m}, skipping"
```

Comment:
-1 to using ```p``` here. Based on this: http://ruby-doc.org/core-2.3.0/Kernel.html#method-i-p it looks like it does an inspect, I think it would be better if it wasn't just a string you were printing. I'd change this to ```puts```.